sdk-python/awaazde/utils.py

```diff
@@ -0,0 +1,137 @@
+import csv
+from datetime import datetime
+
+from constants import CommonConstants
+
+from . import AwaazDeAPI
+
+
+class ADMessageUtils(object):
+    def __init__(self, org, user, password):
+        """
+        Initialize Awaazde Authentication and Call Manager
+        :return:
+        """
+        self.awaazde_api = AwaazDeAPI(org, user, password)
+
+    def get_messages(self, filters):
+        """
+            Gets all messages from awaazde API based on the filters passed
+        """
+
+        def _get_url_params(url):
+            """
+            :param url: A url which looks something like "https://api.awaaz.de/awaazde/v1/xact/message/?state=6&page=2"
+            :returns A dictionary of request params in the url something like
+                    {
+                        "state": 6,
+                        "page": 2
+                    }
+            """
+            all_param_strings = url.split('?', 1)[1].split('&')
+            url_params = {}
+            for param in all_param_strings:
+                url_params[param.split('=', 1)[0]] = param.split('=', 1)[1]
+            return url_params
+
+        # Get next page URL
+        next_page_url = None
+        calls_data = []
+        while True:
+            if not next_page_url:
+                response = self.awaazde_api.messages.list(params=filters)
+            else:
+                # GEts next page number
+                filters['page'] = _get_url_params(next_page_url)['page']
+                # And then we request for the data on the next page
+                response = self.awaazde_api.messages.list(params=filters)
+            if response:
+                calls_data.extend(response['results'])
+                if response['next']:
+                    next_page_url = response['next']
+                else:
+                    break
+            else:
+                print "Error in Fetching Results"
+
+                return calls_data
+        return calls_data
+
+    def schedule_bulk_messages(self, messages):
+        """
+            Sends request to awaazde for scheduling Messages in bulk
+            :param messages: List of messages in the format as follows
+             [{
+                'templatelanguage': 3,
+                'phone_number': '+91999999999',
+                'send_on': "2018-04-04T10:00:00",
+                'values': ['dynamic11', 'dynamic21']
+                },
+                {
+                'templatelanguage': 3,
+                'phone_number': '+919999999998',
+                'values': ['dynamic12', 'dynamic22']
+                }]
+        """
+        messages = {'data': messages}
+        response = self.awaazde_api.messages.bulk_create(messages)
+        return response
+
+    def check_created_message(self, message_data, filter_fields):
```

Comment:
This isn't just checking, and even def of "checking" is ambiguous. I see that this is filtering message data by whether it was created or not, but not even since it is checking by phone number which is error prone.

I think we need a better name and clearer definition. Maybe better to work on this in a document rather than code since I don't see a clear design of this manager.

Reply:
@ashwini-ad comment still applies